catroid/src/createatschool/java/org/catrobat/catroid/createatschool/transfers/FetchTemplatesTask.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.createatschool.transfers;
+
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.os.Build;
+import android.util.Log;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.dialogs.CustomAlertDialogBuilder;
+import org.catrobat.catroid.utils.Utils;
+import org.catrobat.catroid.web.ServerCalls;
+import org.catrobat.catroid.web.WebconnectionException;
+
+public class FetchTemplatesTask extends AsyncTask<Void, Void, String> {
+
+	private static final String TAG = FetchTemplatesTask.class.getSimpleName();
+
+	private Context context;
+	private ProgressDialog progressDialog;
+	private OnFetchTemplatesCompleteListener onFetchTemplatesCompleteListener;
+	private WebconnectionException exception;
+	private String message;
+
+	public FetchTemplatesTask(Context activity) {
+		this.context = activity;
+	}
+
+	public void setOnFetchTemplatesCompleteListener(OnFetchTemplatesCompleteListener listener) {
+		onFetchTemplatesCompleteListener = listener;
+	}
+
+	@Override
+	protected void onPreExecute() {
+		super.onPreExecute();
+		if (context == null) {
+			return;
+		}
+		String title = context.getString(R.string.please_wait);
+		String message = context.getString(R.string.fetching_templates);
+		progressDialog = ProgressDialog.show(context, title, message);
+	}
+
+	@Override
+	protected String doInBackground(Void... arg0) {
+		try {
+			if (!Utils.isNetworkAvailable(context)) {
+				exception = new WebconnectionException(WebconnectionException.ERROR_NETWORK, "Network not available!");
+				return null;
+			}
+
+			return ServerCalls.getInstance().fetchTemplatesList();
+		} catch (WebconnectionException webconnectionException) {
+			Log.e(TAG, Log.getStackTraceString(webconnectionException));
+			message = webconnectionException.getMessage();
+		}
+		return null;
+	}
+
+	@Override
+	protected void onPostExecute(String templatesList) {
+		super.onPostExecute(templatesList);
+
+		if (progressDialog != null && progressDialog.isShowing() && !((Activity) context).isFinishing()) {
+			if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1 && !((Activity) context).isDestroyed()) {
+				progressDialog.dismiss();
+			} else {
+				progressDialog.dismiss();
+			}
+		}
+
+		if (Utils.checkForNetworkError(exception)) {
```

Comment:
To check whether to display an error message or not. 
Maybe it can be refactored somehow, but, again, this is the same implementation as in all other Tasks.